src/NBitcoin/Protocol/InventoryVector.cs

```diff
@@ -22,56 +16,48 @@ public enum InventoryType : uint
 		MSG_WITNESS_TX = MSG_TX | MSG_WITNESS_FLAG,
 		MSG_FILTERED_WITNESS_BLOCK = MSG_FILTERED_BLOCK | MSG_WITNESS_FLAG
 	}
+
 	public class InventoryVector : Payload, IBitcoinSerializable
 	{
-		uint type;
-		uint256 hash = uint256.Zero;
+		private uint type;
+        public InventoryType Type
+        {
+            get
+            {
+                return (InventoryType)this.type;
+            }
+            set
+            {
+                this.type = (uint)value;
+            }
+        }
 
-		public InventoryVector()
-		{
+        private uint256 hash = uint256.Zero;
```

Comment:
same here, can we have the property defined like we like (without the backing field)?

Reply:
Yeah you can't change it, `ref` will work on the same copy that was passed to the method.
This allows to use the serializer as both read and write